projects/demo/src/modules/components/editor-new/examples/6/index.ts

```diff
@@ -0,0 +1,62 @@
+import {Component, Injector, ViewEncapsulation} from '@angular/core';
+import {FormControl} from '@angular/forms';
+import {changeDetection} from '@demo/emulate/change-detection';
+import {
+    tiptapEditorStyles,
+    TUI_EDITOR_EXTENSIONS,
+    TUI_EDITOR_STYLES,
+    TuiEditorTool,
+} from '@taiga-ui/addon-editor';
+import {TuiDestroyService} from '@taiga-ui/cdk';
+
+@Component({
+    selector: 'tui-editor-example-6',
+    templateUrl: './index.html',
+    styleUrls: ['./index.less'],
+    providers: [
+        TuiDestroyService,
+        {
+            provide: TUI_EDITOR_STYLES,
+            useValue: tiptapEditorStyles,
+        },
+        {
+            provide: TUI_EDITOR_EXTENSIONS,
+            deps: [Injector],
+            useFactory: (injector: Injector) => [
+                import('@tiptap/starter-kit').then(({default: module}) => module),
+                import('@tiptap/extension-placeholder').then(({Placeholder}) =>
+                    Placeholder.configure({
+                        placeholder: 'Write something …',
+                        includeChildren: true,
+                    }),
+                ),
+                import('@taiga-ui/addon-editor/extensions/group').then(
+                    ({createGroupExtension}) => createGroupExtension(injector),
```

Comment:
not sure about it's better to configure by injector

Reply:
I think it's better to configure via options like other extensions

Reply:
dont forget to update this page

https://taiga-ui.dev/components/editor-new/DI_tokens

Reply:
fixed